app/frontend/actions/loading.ts

```diff
@@ -0,0 +1,32 @@
+import {Store, State} from '../state'
+
+export default ({setState}: Store) => {
+  return {
+    // TODO: remove "optionalArgsObj", as "loadingAction" should not be responsible for setting
+    // unrelated parts of the state
+    loadingAction: (state: State, message: string, optionalArgsObj?: any) => {
```

Comment:
Do you like defining the functions in the return statement more than having them defined like this? 
```
const someAction= () => {someFunctionBody}
return {
 someAction,
}
```
I probably find the second approach a bit more readable but maybe there is a advantage to you approach I am not seeing.

Reply:
Good point :+1: will change that

Reply:
@PeterBenc fixed in the last comment & rebased

Reply:
Hmm some parts, were not rebased correctly will fix this

Reply:
Actually, should be fine.